src/python/zquantum/core/interfaces/optimizer.py

```diff
@@ -105,3 +106,50 @@ def construct_history_info(cost_function, keep_history):
     if keep_history and hasattr(cost_function, "gradient"):
         histories["gradient_history"] = cost_function.gradient.history
     return histories
+
+
+class MetaOptimizer(ABC):
+    def __init__(
+        self,
+        ansatz: Ansatz,
+        inner_optimizer: Optimizer,
+        estimation_tasks_factory: Callable[[Any], EstimationTasksFactory],
+        cost_function_factory: Callable[[EstimationTasksFactory], CostFunction],
+        recorder: RecorderFactory = _recorder,
+    ) -> None:
+        """
+        Optimizers that modify cost function throughout optimization.
+        See RQAOA (in zquantum.qaoa) or LayerwiseAnsatzOptimizer (in
```

Comment:
We should not refer to packages that we depend on, because if something changes in `zquantum.qaoa`, then  `core` wouldn't know that.
I'd suggest creating a slightly more involved `MockMetaOptimizer` that could be referenced here and could be treated as an example.